response_writer.go

```diff
@@ -425,3 +426,30 @@ func (w *responseWriter) Push(target string, opts *http.PushOptions) error {
 		return err
 	}
 }
+
+type extWriter struct {
+	rw *responseWriter
+}
+
+func (w *extWriter) Write(p []byte) (n int, err error) {
+	return w.rw.Write(p)
```

Comment:
When to write header frame? Since this `extWriter` has already hijack the whole response writing process (header & body).

Reply:
Got it, processed in `w.rw.Write(p)`.